src/formio/components/Select.js

```diff
@@ -115,6 +116,16 @@ class Select extends Formio.Components.components.select {
     }
   }
 
+  beforeSubmit() {
+      if (this.component.appointmentsShowProducts || this.component.appointmentsShowLocations) {
+        // For these two types of components we need to send both the identifier and name to the backend
+        const value = this._data[this.component.key].toString();
+        const selectedOption = this.selectOptions.filter(option => option.value === value)[0];
```

Comment:
From what I could see when changing the selected option in the html it did not do anything to change the component itself, just changed what is in `this._data` so this seemed to be the best way to access the data that we would want.